Release plan, update with new branch names and release notes

s-canchi opened the pull request:
## PR type

- [ ] New tutorial
- [x] Fixes to tutorials only on latest
- [ ] Fixes to tutorials on stable and latest
- [ ] Feature enhancement
- [ ] Tutorial formatting
- [ ] Website formatting
- [x] Other

## PR Checklist

- [ ] Release label added
- [x] PR category label added
- [ ] PR mergeable
- [x] Clean build logs
- [ ] Linked related issues

## PR Description
Release plan documentation along with release notes for the website are added. In addition, the automated workflows have been updated to reflect the new branching scheme. The PULL REQUEST TEMPLATE has also been updated to reflect the new branching name and with added reviewer checklist. 

## Review format
Review in comments.

## Timeline
Time sensitive since workflows needs to be merged for alpha test. Planned test on Sept 14th, 2020.

s-canchi committed:
Modified workflows with updated branch naming scheme

s-canchi committed:
Added releaseplan and release notes for repo

s-canchi committed:
"Updating loacl branch"
Merge remote-tracking branch 'origin/dev' into scanchi-releaseplan-fixes

s-canchi committed:
Updated release notes

s-canchi committed:
Updated release plan with workflow discussion

s-canchi committed:
Added release workflow flow diagram

marisalim commented:
@s-canchi the PR template still has the `latest` branch instead of `dev`. maybe you can edit the template on this PR?

s-canchi committed:
Updated PULL REQUEST TEMPLATE with branch names and reviewer checklist

s-canchi committed:
Corrected to capital letter in PULL REQUEST TEMPLATE

jeremywalter reviewed (approved):
I don't see any fundamental problems...
but these things irked me a little:
+ New --> maybe call it New Releases?
+ Maintenance --> seems more like **Updates** or **Updates and Fixes** (but that might just be me)
                      --> the links go to the updated lesson which seems kind of strange context-wise for me... like... when I click on that I expect to see the changes or more explanation for them... with the links I'd want the link to be the title of the lesson with the description so:
[Enabling User Choices with Javascript]: added additional context
--note this lesson has a different title in the sidebar, dunno if that's being sorted elsewhere

+ Updated On --> the header... title of this page made kind of look around for what was updated... I don't know if it matters, but it would make more sense to me on the page if it was 'Latest Release' and then "Updated on" as a sub-title.
Seems to really be coming together. I like the release notes and if I need to go tweak them for the public facing version I think it makes the job relatively easy.

s-canchi committed:
Updated release notes with lesson names, description and heading

s-canchi committed:
Merge branch 'dev' into scanchi-releaseplan-fixes

s-canchi committed:
Updating with bold font for date

s-canchi committed:
Merging latest changes to dev
Merge branch 'scanchi-releaseplan-fixes' of https://github.com/nih-cfde/training-and-engagement into scanchi-releaseplan-fixes
